Automattic/wp-calypso
# media: migrate `editMedia` to use React Query

This PR is part of a series of PRs to migrate the media library to React Query. I'm going to tackle mutations first and integrate them intermediately with state which currently still is in Redux. That way we'll get rid of all the Redux related code step by step similar to how we migrated the media library from Flux to Redux.

#### Changes proposed in this Pull Request

* media: migrate `editMedia` to use React Query
* Remove Redux action creators, thunks and data layer parts related to `editMedia`

#### Testing instructions

Please note that `editMedia` is responsible only for image manipulation (rotation, flips, etc).

1. Go to `/media`, select one or more images and click _Edit_
2. Hit _Edit Image_ and do some changes to the image itself (crop, rotation, etc), then hit _Done_ 

Now a few things should be observable:

* the selected image gets updated optimistically (using a transient item showing a spinner)
* a network request to `/sites/<siteId>/media/<mediaId>/edit`

After the request succeeded, the transient image should be replaced with the actual image response from the server.

Now go to 1. again but hit _Restore Original_ and make sure that works as well.

In case of an error (block the network request in your browser devtools) you should see an error notice and the optimistically updated image should be reverted to the previous state.

Additionally, this should be tested from the media modal used in the editor as well.

Related to #58574
